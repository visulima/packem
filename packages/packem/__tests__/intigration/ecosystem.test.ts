import { readdirSync } from "node:fs";
import { rm } from "node:fs/promises";

import { readFileSync } from "@visulima/fs";
import { join } from "@visulima/path";
import { afterEach, describe, expect, it } from "vitest";

import { createPackemConfig, execPackem } from "../helpers";

const ecosystemPath = join(__dirname, "../..", "__fixtures__", "ecosystem");

// eslint-disable-next-line security/detect-non-literal-fs-filename
const ecosystemSuites = readdirSync(ecosystemPath, { withFileTypes: true })
    .filter((dirent) => dirent.isDirectory())
    .map((dirent) => dirent.name);

describe("packem ecosystem", () => {
    afterEach(async () => {
        for await (const suite of ecosystemSuites) {
            await rm(join(ecosystemPath, suite, "dist"), { force: true, recursive: true });
        }
    });

    it.each(ecosystemSuites)("should work with provided '%s' ecosystem suite", async (suite) => {
        expect.assertions(3);

        const fullSuitePath = join(ecosystemPath, suite);

        await createPackemConfig(fullSuitePath, {
            isolatedDeclarationTransformer: "typescript",
            transformer: "esbuild",
        });

        const binProcess = await execPackem("build", [], {
            cwd: fullSuitePath,
        });

        expect(binProcess.stderr).toBe("");
        expect(binProcess.exitCode).toBe(0);

        // eslint-disable-next-line security/detect-non-literal-fs-filename
        const distributionFiles = readdirSync(join(fullSuitePath, "dist"), {
            recursive: true,
            withFileTypes: true,
        })
            .filter((dirent) => dirent.isFile())

            .map((dirent) => readFileSync(join(dirent.parentPath, dirent.name)));

        expect(distributionFiles).toMatchSnapshot();
    });

    it.todo.each(ecosystemSuites)("should work with provided '%s' ecosystem suite and oxc resolver", async (suite) => {
        const fullSuitePath = join(ecosystemPath, suite);

        await createPackemConfig(fullSuitePath, {
            experimental: {
                oxcResolve: true,
            },
            isolatedDeclarationTransformer: "typescript",
            transformer: "esbuild",
        });

        const binProcess = await execPackem("build", [], {
            cwd: fullSuitePath,
        });

        expect(binProcess.stderr).toBe("");
        expect(binProcess.exitCode).toBe(0);

        const distributionFiles = readdirSync(join(fullSuitePath, "dist"), {
            recursive: true,
            withFileTypes: true,
        })
            .filter((dirent) => dirent.isFile())
            .map((dirent) => readFileSync(join(dirent.parentPath, dirent.name)));

        expect(distributionFiles).toMatchSnapshot();
    });
});
