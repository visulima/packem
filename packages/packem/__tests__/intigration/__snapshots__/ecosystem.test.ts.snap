// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`packem ecosystem > should work with provided 'advanced-tailwind' ecosystem suite 1`] = `
[
  "import "./styles.css";
export interface AdvancedTailwindComponent {
    name: string;
    className: string;
    description: string;
}
export declare const components: AdvancedTailwindComponent[];
export declare function getComponent(name: string): AdvancedTailwindComponent | undefined;
export declare function getAllComponents(): AdvancedTailwindComponent[];
//# sourceMappingURL=index.d.ts.map
",
  "{"file":"index.d.ts","mappings":"AACA,OAAO,cAAc,CAAC;AAGtB,MAAM,WAAW,yBAAyB;IACtC,IAAI,EAAE,MAAM,CAAC;IACb,SAAS,EAAE,MAAM,CAAC;IAClB,WAAW,EAAE,MAAM,CAAC;CACvB;AAED,eAAO,MAAM,UAAU,EAAE,yBAAyB,EAgBjD,CAAC;AAEF,wBAAgB,YAAY,CAAC,IAAI,EAAE,MAAM,GAAG,yBAAyB,GAAG,SAAS,CAEhF;AAED,wBAAgB,gBAAgB,IAAI,yBAAyB,EAAE,CAE9D","names":[],"sourceRoot":"","sources":["../src/index.ts"],"version":3}",
  "import { cssStyleInject } from '@visulima/css-style-inject';

var css = "/*! tailwindcss v4.1.12 | MIT License | https://tailwindcss.com */\\n@layer theme, base, components, utilities;\\n@layer theme {\\n  :root, :host {\\n    --font-sans: ui-sans-serif, system-ui, sans-serif, \\"Apple Color Emoji\\",\\n      \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\";\\n    --font-mono: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, \\"Liberation Mono\\",\\n      \\"Courier New\\", monospace;\\n    --default-font-family: var(--font-sans);\\n    --default-mono-font-family: var(--font-mono);\\n  }\\n}\\n@layer base {\\n  *, ::after, ::before, ::backdrop, ::file-selector-button {\\n    box-sizing: border-box;\\n    margin: 0;\\n    padding: 0;\\n    border: 0 solid;\\n  }\\n  html, :host {\\n    line-height: 1.5;\\n    -webkit-text-size-adjust: 100%;\\n    tab-size: 4;\\n    font-family: var(--default-font-family, ui-sans-serif, system-ui, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\");\\n    font-feature-settings: var(--default-font-feature-settings, normal);\\n    font-variation-settings: var(--default-font-variation-settings, normal);\\n    -webkit-tap-highlight-color: transparent;\\n  }\\n  hr {\\n    height: 0;\\n    color: inherit;\\n    border-top-width: 1px;\\n  }\\n  abbr:where([title]) {\\n    -webkit-text-decoration: underline dotted;\\n    text-decoration: underline dotted;\\n  }\\n  h1, h2, h3, h4, h5, h6 {\\n    font-size: inherit;\\n    font-weight: inherit;\\n  }\\n  a {\\n    color: inherit;\\n    -webkit-text-decoration: inherit;\\n    text-decoration: inherit;\\n  }\\n  b, strong {\\n    font-weight: bolder;\\n  }\\n  code, kbd, samp, pre {\\n    font-family: var(--default-mono-font-family, ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, \\"Liberation Mono\\", \\"Courier New\\", monospace);\\n    font-feature-settings: var(--default-mono-font-feature-settings, normal);\\n    font-variation-settings: var(--default-mono-font-variation-settings, normal);\\n    font-size: 1em;\\n  }\\n  small {\\n    font-size: 80%;\\n  }\\n  sub, sup {\\n    font-size: 75%;\\n    line-height: 0;\\n    position: relative;\\n    vertical-align: baseline;\\n  }\\n  sub {\\n    bottom: -0.25em;\\n  }\\n  sup {\\n    top: -0.5em;\\n  }\\n  table {\\n    text-indent: 0;\\n    border-color: inherit;\\n    border-collapse: collapse;\\n  }\\n  :-moz-focusring {\\n    outline: auto;\\n  }\\n  progress {\\n    vertical-align: baseline;\\n  }\\n  summary {\\n    display: list-item;\\n  }\\n  ol, ul, menu {\\n    list-style: none;\\n  }\\n  img, svg, video, canvas, audio, iframe, embed, object {\\n    display: block;\\n    vertical-align: middle;\\n  }\\n  img, video {\\n    max-width: 100%;\\n    height: auto;\\n  }\\n  button, input, select, optgroup, textarea, ::file-selector-button {\\n    font: inherit;\\n    font-feature-settings: inherit;\\n    font-variation-settings: inherit;\\n    letter-spacing: inherit;\\n    color: inherit;\\n    border-radius: 0;\\n    background-color: transparent;\\n    opacity: 1;\\n  }\\n  :where(select:is([multiple], [size])) optgroup {\\n    font-weight: bolder;\\n  }\\n  :where(select:is([multiple], [size])) optgroup option {\\n    padding-inline-start: 20px;\\n  }\\n  ::file-selector-button {\\n    margin-inline-end: 4px;\\n  }\\n  ::placeholder {\\n    opacity: 1;\\n  }\\n  @supports (not (-webkit-appearance: -apple-pay-button))  or (contain-intrinsic-size: 1px) {\\n    ::placeholder {\\n      color: currentcolor;\\n      @supports (color: color-mix(in lab, red, red)) {\\n        color: color-mix(in oklab, currentcolor 50%, transparent);\\n      }\\n    }\\n  }\\n  textarea {\\n    resize: vertical;\\n  }\\n  ::-webkit-search-decoration {\\n    -webkit-appearance: none;\\n  }\\n  ::-webkit-date-and-time-value {\\n    min-height: 1lh;\\n    text-align: inherit;\\n  }\\n  ::-webkit-datetime-edit {\\n    display: inline-flex;\\n  }\\n  ::-webkit-datetime-edit-fields-wrapper {\\n    padding: 0;\\n  }\\n  ::-webkit-datetime-edit, ::-webkit-datetime-edit-year-field, ::-webkit-datetime-edit-month-field, ::-webkit-datetime-edit-day-field, ::-webkit-datetime-edit-hour-field, ::-webkit-datetime-edit-minute-field, ::-webkit-datetime-edit-second-field, ::-webkit-datetime-edit-millisecond-field, ::-webkit-datetime-edit-meridiem-field {\\n    padding-block: 0;\\n  }\\n  ::-webkit-calendar-picker-indicator {\\n    line-height: 1;\\n  }\\n  :-moz-ui-invalid {\\n    box-shadow: none;\\n  }\\n  button, input:where([type=\\"button\\"], [type=\\"reset\\"], [type=\\"submit\\"]), ::file-selector-button {\\n    appearance: button;\\n  }\\n  ::-webkit-inner-spin-button, ::-webkit-outer-spin-button {\\n    height: auto;\\n  }\\n  [hidden]:where(:not([hidden=\\"until-found\\"])) {\\n    display: none !important;\\n  }\\n}\\n@layer utilities;\\n@layer base {\\n  [type='text'],input:where(:not([type])),[type='email'],[type='url'],[type='password'],[type='number'],[type='date'],[type='datetime-local'],[type='month'],[type='search'],[type='tel'],[type='time'],[type='week'],[multiple],textarea,select {\\n    appearance: none;\\n    background-color: #fff;\\n    border-color: oklch(55.1% 0.027 264.364);\\n    border-width: 1px;\\n    border-radius: 0px;\\n    padding-top: 0.5rem;\\n    padding-right: 0.75rem;\\n    padding-bottom: 0.5rem;\\n    padding-left: 0.75rem;\\n    font-size: 1rem;\\n    line-height: 1.5rem;\\n    --tw-shadow: 0 0 #0000;\\n    &:focus {\\n      outline: 2px solid transparent;\\n      outline-offset: 2px;\\n      --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);\\n      --tw-ring-offset-width: 0px;\\n      --tw-ring-offset-color: #fff;\\n      --tw-ring-color: oklch(54.6% 0.245 262.881);\\n      --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);\\n      --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color);\\n      box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow);\\n      border-color: oklch(54.6% 0.245 262.881);\\n    }\\n  }\\n  input::placeholder,textarea::placeholder {\\n    color: oklch(55.1% 0.027 264.364);\\n    opacity: 1;\\n  }\\n  ::-webkit-datetime-edit-fields-wrapper {\\n    padding: 0;\\n  }\\n  ::-webkit-date-and-time-value {\\n    min-height: 1.5em;\\n  }\\n  ::-webkit-date-and-time-value {\\n    text-align: inherit;\\n  }\\n  ::-webkit-datetime-edit {\\n    display: inline-flex;\\n  }\\n  ::-webkit-datetime-edit,::-webkit-datetime-edit-year-field,::-webkit-datetime-edit-month-field,::-webkit-datetime-edit-day-field,::-webkit-datetime-edit-hour-field,::-webkit-datetime-edit-minute-field,::-webkit-datetime-edit-second-field,::-webkit-datetime-edit-millisecond-field,::-webkit-datetime-edit-meridiem-field {\\n    padding-top: 0;\\n    padding-bottom: 0;\\n  }\\n  select {\\n    background-image: url(\\"data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 20 20'%3e%3cpath stroke='oklch(55.1%25 0.027 264.364)' stroke-linecap='round' stroke-linejoin='round' stroke-width='1.5' d='M6 8l4 4 4-4'/%3e%3c/svg%3e\\");\\n    background-position: right 0.5rem center;\\n    background-repeat: no-repeat;\\n    background-size: 1.5em 1.5em;\\n    padding-right: 2.5rem;\\n    print-color-adjust: exact;\\n  }\\n  [multiple],[size]:where(select:not([size=\\"1\\"])) {\\n    background-image: initial;\\n    background-position: initial;\\n    background-repeat: unset;\\n    background-size: initial;\\n    padding-right: 0.75rem;\\n    print-color-adjust: unset;\\n  }\\n  [type='checkbox'],[type='radio'] {\\n    appearance: none;\\n    padding: 0;\\n    print-color-adjust: exact;\\n    display: inline-block;\\n    vertical-align: middle;\\n    background-origin: border-box;\\n    user-select: none;\\n    flex-shrink: 0;\\n    height: 1rem;\\n    width: 1rem;\\n    color: oklch(54.6% 0.245 262.881);\\n    background-color: #fff;\\n    border-color: oklch(55.1% 0.027 264.364);\\n    border-width: 1px;\\n    --tw-shadow: 0 0 #0000;\\n  }\\n  [type='checkbox'] {\\n    border-radius: 0px;\\n  }\\n  [type='radio'] {\\n    border-radius: 100%;\\n  }\\n  [type='checkbox']:focus,[type='radio']:focus {\\n    outline: 2px solid transparent;\\n    outline-offset: 2px;\\n    --tw-ring-inset: var(--tw-empty,/*!*/ /*!*/);\\n    --tw-ring-offset-width: 2px;\\n    --tw-ring-offset-color: #fff;\\n    --tw-ring-color: oklch(54.6% 0.245 262.881);\\n    --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);\\n    --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);\\n    box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow);\\n  }\\n  [type='checkbox']:checked,[type='radio']:checked {\\n    border-color: transparent;\\n    background-color: currentColor;\\n    background-size: 100% 100%;\\n    background-position: center;\\n    background-repeat: no-repeat;\\n  }\\n  [type='checkbox']:checked {\\n    background-image: url(\\"data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3cpath d='M12.207 4.793a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0l-2-2a1 1 0 011.414-1.414L6.5 9.086l4.293-4.293a1 1 0 011.414 0z'/%3e%3c/svg%3e\\");\\n    @media (forced-colors: active) {\\n      appearance: auto;\\n    }\\n  }\\n  [type='radio']:checked {\\n    background-image: url(\\"data:image/svg+xml,%3csvg viewBox='0 0 16 16' fill='white' xmlns='http://www.w3.org/2000/svg'%3e%3ccircle cx='8' cy='8' r='3'/%3e%3c/svg%3e\\");\\n    @media (forced-colors: active) {\\n      appearance: auto;\\n    }\\n  }\\n  [type='checkbox']:checked:hover,[type='checkbox']:checked:focus,[type='radio']:checked:hover,[type='radio']:checked:focus {\\n    border-color: transparent;\\n    background-color: currentColor;\\n  }\\n  [type='checkbox']:indeterminate {\\n    background-image: url(\\"data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 16 16'%3e%3cpath stroke='white' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M4 8h8'/%3e%3c/svg%3e\\");\\n    border-color: transparent;\\n    background-color: currentColor;\\n    background-size: 100% 100%;\\n    background-position: center;\\n    background-repeat: no-repeat;\\n    @media (forced-colors: active) {\\n      appearance: auto;\\n    }\\n  }\\n  [type='checkbox']:indeterminate:hover,[type='checkbox']:indeterminate:focus {\\n    border-color: transparent;\\n    background-color: currentColor;\\n  }\\n  [type='file'] {\\n    background: unset;\\n    border-color: inherit;\\n    border-width: 0;\\n    border-radius: 0;\\n    padding: 0;\\n    font-size: unset;\\n    line-height: inherit;\\n  }\\n  [type='file']:focus {\\n    outline: 1px solid ButtonText;\\n    outline: 1px auto -webkit-focus-ring-color;\\n  }\\n}\\n";
cssStyleInject(css,{});

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
const components = [
  {
    name: "Primary Button",
    className: "btn-primary",
    description: "A primary button with hover effects and transitions"
  },
  {
    name: "Card",
    className: "card",
    description: "A card component with shadow and border styling"
  },
  {
    name: "Gradient Text",
    className: "text-gradient",
    description: "Text with blue to purple gradient effect"
  }
];
function getComponent(name) {
  return components.find((comp) => comp.name === name);
}
__name(getComponent, "getComponent");
function getAllComponents() {
  return components;
}
__name(getAllComponents, "getAllComponents");

export { components, getAllComponents, getComponent };
//# sourceMappingURL=index.js.map
",
  "{"version":3,"file":"index.js","sources":["../src/styles.css","../src/index.ts"],"sourcesContent":["@import \\"tailwindcss\\";\\n\\n@source \\"../node_modules/preline/dist/*.js\\";\\n@import \\"preline/variants.css\\";\\n\\n/* Optional Preline UI Datepicker Plugin */\\n/* @import \\"preline/src/plugins/datepicker/styles.css\\"; */\\n\\n/* Plugins */\\n@plugin '@tailwindcss/typography';\\n@plugin '@tailwindcss/forms';\\n\\n@custom-variant dark (&:where(.dark, .dark *));\\n","// Import the advanced Tailwind CSS styles\\nimport \\"./styles.css\\";\\n\\n// Export a simple component interface for testing\\nexport interface AdvancedTailwindComponent {\\n    name: string;\\n    className: string;\\n    description: string;\\n}\\n\\nexport const components: AdvancedTailwindComponent[] = [\\n    {\\n        name: \\"Primary Button\\",\\n        className: \\"btn-primary\\",\\n        description: \\"A primary button with hover effects and transitions\\"\\n    },\\n    {\\n        name: \\"Card\\",\\n        className: \\"card\\",\\n        description: \\"A card component with shadow and border styling\\"\\n    },\\n    {\\n        name: \\"Gradient Text\\",\\n        className: \\"text-gradient\\",\\n        description: \\"Text with blue to purple gradient effect\\"\\n    }\\n];\\n\\nexport function getComponent(name: string): AdvancedTailwindComponent | undefined {\\n    return components.find(comp => comp.name === name);\\n}\\n\\nexport function getAllComponents(): AdvancedTailwindComponent[] {\\n    return components;\\n}\\n"],"names":[],"mappings":";;AAAA,IAAA,GAAA,GAAA,80UAAyC;cAGxB,CAAA,GAAA,CAAA,EAAA,CAAA;;;;ACOV,MAAM,UAAA,GAA0C;AAAA,EACnD;AAAA,IACI,IAAA,EAAM,gBAAA;AAAA,IACN,SAAA,EAAW,aAAA;AAAA,IACX,WAAA,EAAa;AAAA,GACjB;AAAA,EACA;AAAA,IACI,IAAA,EAAM,MAAA;AAAA,IACN,SAAA,EAAW,MAAA;AAAA,IACX,WAAA,EAAa;AAAA,GACjB;AAAA,EACA;AAAA,IACI,IAAA,EAAM,eAAA;AAAA,IACN,SAAA,EAAW,eAAA;AAAA,IACX,WAAA,EAAa;AAAA;AAErB;AAEO,SAAS,aAAa,IAAA,EAAqD;AAC9E,EAAA,OAAO,UAAA,CAAW,IAAA,CAAK,CAAA,IAAA,KAAQ,IAAA,CAAK,SAAS,IAAI,CAAA;AACrD;AAFgB,MAAA,CAAA,YAAA,EAAA,cAAA,CAAA;AAIT,SAAS,gBAAA,GAAgD;AAC5D,EAAA,OAAO,UAAA;AACX;AAFgB,MAAA,CAAA,gBAAA,EAAA,kBAAA,CAAA;;;;"}",
  "declare var css: string;
export default css;
export { css };
//# sourceMappingURL=styles.d.ts.map
",
  "{"file":"styles.d.ts","mappings":"AACA,QAAA,IAAI,GAAG,QAAi1U,CAAC;AAIz1U,eAAe,GAAG,CAAC;AAEnB,OAAO,EACL,GAAG,EACJ,CAAC","names":[],"sourceRoot":"","sources":["../src/styles.css"],"version":3}",
]
`;
